lib/SPIRV/SPIRVLowerBitCastToNonStandardType.cpp

```diff
@@ -204,6 +204,7 @@ SPIRVLowerBitCastToNonStandardTypePass::run(Function &F,
     // that's what we got, it's still a dead instruction.
     if (VH.pointsToAliveValue() && !isa<UndefValue>(VH)) {
       auto *VT = dyn_cast<VectorType>(VH->getType());
+      assert(VT && "must have vector type");
```

Comment:
Just `dyn_cast` -> `cast` would be enough. We should never use `dyn_cast` if it is not followed by `if (DynCasted != nullptr)`, because it doesn't make sense. `cast` contains all necessary assertions within it

Reply:
done :)